packages/transaction-manager/test/index.test.ts

```diff
@@ -339,6 +355,165 @@ describe('index', () => {
     });
   });
 
+  describe('persistTransaction - InMemoryTransactionManager', () => {
+    describe('in a new channel', () => {
+      it('can create a clear in memory transaction in a new channel', async () => {
+        const transactionManager = new InMemoryTransactionManager(fakeDataAccess);
+
+        const ret = await transactionManager.persistTransaction(data, channelId, extraTopics);
+
+        expect(ret.result).toEqual({});
+        expect(ret.meta).toEqual({
+          dataAccessMeta: { persisted: false },
+          encryptionMethod: undefined,
+        });
+        expect(ret).toHaveProperty('inMemoryTransaction');
+        expect(fakeDataAccess.persistTransaction).not.toHaveBeenCalled();
+      });
+
+      it('can persist an encrypted transaction in a new channel', async () => {
+        const transactionManager = new InMemoryTransactionManager(fakeDataAccess);
+
+        const ret = await transactionManager.persistTransaction(data, channelId, extraTopics, [
+          TestData.idRaw1.encryptionParams,
+          TestData.idRaw2.encryptionParams,
+          TestData.idRaw3.encryptionParams,
+        ]);
+
+        expect(ret.result).toEqual({});
+        expect(ret.meta).toEqual({
+          dataAccessMeta: { persisted: false },
+          encryptionMethod: 'ecies-aes256-gcm',
+        });
+        expect(ret).toHaveProperty('inMemoryTransaction');
+        expect(fakeDataAccess.persistTransaction).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('in an existing new channel', () => {
+      afterEach(() => {
+        jest.clearAllMocks();
+      });
+      it('can persist a clear transaction in an existing channel', async () => {
+        const transactionManager = new InMemoryTransactionManager(fakeDataAccess);
+
+        const ret = await transactionManager.persistTransaction(data2, channelId, extraTopics);
```

Comment:
Unfortunately, I think the other test for persisting clear transaction to an existing channel has the same error.

Reply:
In the case of a new channel we use `channelId` which is derived from the `data` hash, that way `process-transaction` method would have `channelId` and `hash` as equals leading to creation of a new channel.

In the case of an existing channel, we use `data2` instead of `data` leading to the `else` logic in `process-transaction` method, we also mock `getTransactionsByChannelId` to return fake transactions. Leading to the transaction being added to the existing channel.